traits.yml

```diff
@@ -84,10 +145,210 @@ traits:
+          IES-LM-63, see:
+
+              https://store.ies.org/product/lm-63-19-approved-method-ies-standard-file-format-for-the-electronic-transfer-of-photometric-data-and-related-information
+      SourcePath:
+        description: >
+          Defines a singular path within the entity's scene
+          hierarchy that should be used as the root, ignoring any other
+          peer paths.
+
+
+          Paths are delimited by a forward slash (/) and are always
+          considered absolute from the root of the entities scene
+          hierarchy.
+
+
+          For example "/cameras/1103/camera"
+      SourcePaths:
+        description: >
+          Defines the root paths within the entity data that form ths
+          start of a discreet hierarchy, labelled by the property key.
+
+
+          For example:
+            left: "/cameras/1103/camera/left"
+            right: "/cameras/1103/camera/right"
+  timeDomain:
+    description: >
+      Traits that are concerned with describing how data changes or is
+      parameterised over time.
+    members:
+      FrameRanged:
+        description: The data varies over a number of discrete frames.
+        usage:
+          - entity
+        properties:
+          startFrame:
+            description: >
+              The first frame covered by the entity data (inclusive).
+            type: integer
+          endFrame:
+            description: >
+              The last frame covered by the entity data (inclusive).
+            type: integer
+          inFrame:
+            description: >
+              The first frame to be used within startFrame/endFrame
+              (inclusive).
+            type: integer
+          outFrame:
+            description: >
+              The last frame to be used within startFrame/endFrame
+              (inclusive).
+            type: integer
+          framesPerSecond:
+            description: >
+              The number of frames that occur within one second.
+            type: float
+          step:
+            description: >
+              The interval between frames of the entity's data. A
+              startFrame of 21 and a step of 10 would indicate that the
+              entity data only covers frames 21, 31, 41, ...
+            type: integer
+      Static:
+        description: The data does not vary over time.
+        usage:
+          - entity
+  twoDimensional:
+    description: >
+      Traits that are concerned with describing two-dimensional data,
+      more colloquially known as 'images'.
+    members:
+      Image:
+        description: >
+          A family trait that should be composed with more specific
+          traits for any entity that holds two-dimensional data.
+        usage:
+          - entity
+      PixelBased:
+        description: >
+          Entity data is encoded in a pixel grid of two dimensions.
+          Also known as 'raster' data.
+        properties:
+          displayWindowWidth:
```

Comment:
I'm not sure I follow. 

As in, we definitely need width/height, but don't definitely need offset yet? 

Or width/height indeed doesn't make sense without offset, but we wanted to enforce naming of these properties now, so that in the future when it is needed and someone else is in charge of adding these properties, we don't choose the wrong name?

Reply:
Both. We need width/height/pixelAspect now. We _know_ that overscan is a thing, and some people define it in different ways, etc... so we'd need to pick a representation that makes sense. As overscan _is_ a thing, then 'width' is ambiguous. Is it the width of the pixel data, or is it the width of the "apparent image". Pixel data is generally never what you really want (eg: openEXR supports the data window being smaller as well as larger than the display window and varies based on compression/authoring), so right now, we need to concretely describe regardless of what extra/in-filled data is in the file, what size should this be presented at. OpenEXR encodes all the relevant data in the file so you can properly load it (minus perhaps PAR, I cant remember). This might not always be the case, and especially for write, we need to define this before an image exists. As such, this is the thing we really need to be able to describe today. It felt like calling it just width and height violates our "says what it is on the tin" rule. Hoped that calling it this (openEXR term) means that when we figure out how we want to describe overscan, then this still makes sense. Eg. one definition I've seen before is t,b,l,r as extra pixels on each edge of the display window, or the same described as a percentage of the displaywindow width or height.  You could also describe the dataWindow directly instead (though as it stands that would assume that the displayWindowWidth/Height is 0,0 origin).

Reply:
Added some notes and clarified https://github.com/OpenAssetIO/OpenAssetIO-MediaCreation/pull/71/commits/bc39a99225ae9bb064b2de5bddd2b66135584c28

Reply:
I understand that width/height is distinct from displayWidth/displayHeight.  Perhaps I'm getting confused what displayWidth/Height means.

Based on the description in the YAML I assumed it was a viewport on the pixel data, i.e. no scaling involved.  

But it sounds from the above that it's more like the width/height that the viewable image (i.e. discarding any overscan pixels) should be stretched/squashed into when displaying?

If that's the case, and it's not just my unique ignorance that misled me, perhaps the description could be finessed?

Reply:
Im really struggling to figure out a better phrasing - can sync up offline. Simplified [here](https://github.com/OpenAssetIO/OpenAssetIO-MediaCreation/pull/71/commits/c8c33be32176c5304c54394090bcffcc166fdf1f)